### PR TITLE
Direct messages

### PR DESCRIPTION
**Pull request by robygonzalez**
Fixes some bugs from the direct messages tutorial

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -19,70 +19,61 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.datastore.FetchOptions;
```
Comment (dgalbraith33) [inline src/main/java/com/google/codeu/data/Datastore.java]: nit: We should leave this import where it was.

File: src/main/webapp/WEB-INF/index.yaml [hunk 2]
```diff
@@ -16,5 +16,6 @@ indexes:
 - kind: Message
   properties:
   - name: user
+  - name: recipient
   - name: timestamp
-    direction: desc
+    direction: desc
```
Comment (dgalbraith33) [inline src/main/webapp/WEB-INF/index.yaml]: nit: It looks like the newline got stripped from this file.

File: src/main/webapp/maps.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
```
Comment (dgalbraith33) [inline src/main/webapp/maps.html]: Lets leave this file out of this PR. You can git rm filename then commit and push.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 4]
```diff
@@ -19,70 +19,61 @@
...
-   * posted matching the query and user. If user is blank, then get messages from all users matching the query.
-   *  Output list is sorted by time, descending order.
+   * Gets messages posted by a specific user.
+   *
+   * @return a list of messages posted by the user, or empty list if user has never posted a
+   *     message. List is sorted by time descending.
    */
-
-  public List<Message> getMessagesByQuery(Query query, String user) {
+  public List<Message> getMessagesByQuery(Query query, String user, String recipient) {
```
Comment (dgalbraith33) [inline src/main/java/com/google/codeu/data/Datastore.java]: Could this just be "public List<Message> getMessagesByQuery(Query query, String recipient) {"?

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 5]
```diff
@@ -19,70 +19,61 @@
...
+    messageEntity.setProperty("user", message.getUser());
+    messageEntity.setProperty("text", message.getText());
+    messageEntity.setProperty("timestamp", message.getTimestamp());
+    messageEntity.setProperty("recipient", message.getRecipient());
+
+    datastore.put(messageEntity);
   }
 
   /**
-   * Gets messages based on a particular query and user (optional).
```
Comment (dgalbraith33) [inline src/main/java/com/google/codeu/data/Datastore.java]: I think we can leave this comment as is probably.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 6]
```diff
@@ -19,70 +19,61 @@
...
-
-  public List<Message> getMessagesByQuery(Query query, String user) {
+  public List<Message> getMessagesByQuery(Query query, String user, String recipient) {
 
     List<Message> messages = new ArrayList<>();
     boolean useAnyMessage = user.isEmpty();
 
-
-    Query query =
+    Query newQuery =
```
Comment (dgalbraith33) [inline src/main/java/com/google/codeu/data/Datastore.java]: It looks like this variable is not used now?

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 4 above)
Comment (aokennard) [reply src/main/java/com/google/codeu/data/Datastore.java]: I made the helper mainly for encapsulating some public feed functions, so we should probably keep it as query, user. It should be fixed on master.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 6 above)
Comment (aokennard) [reply src/main/java/com/google/codeu/data/Datastore.java]: Removed this variable in working version

**Pull request closed without merging**